components/eamxx/src/physics/ace/eamxx_ace_process_interface.cpp

```diff
@@ -0,0 +1,855 @@
+              input_view(0, 27, ilat, ilon) = wind(ilat, ilon, 0, 4);
+              input_view(0, 28, ilat, ilon) = wind(ilat, ilon, 0, 5);
+              input_view(0, 29, ilat, ilon) = wind(ilat, ilon, 0, 6);
+              input_view(0, 30, ilat, ilon) = wind(ilat, ilon, 0, 7);
+              input_view(0, 31, ilat, ilon) = wind(ilat, ilon, 1, 0);
+              input_view(0, 32, ilat, ilon) = wind(ilat, ilon, 1, 1);
+              input_view(0, 33, ilat, ilon) = wind(ilat, ilon, 1, 2);
+              input_view(0, 34, ilat, ilon) = wind(ilat, ilon, 1, 3);
+              input_view(0, 35, ilat, ilon) = wind(ilat, ilon, 1, 4);
+              input_view(0, 36, ilat, ilon) = wind(ilat, ilon, 1, 5);
+              input_view(0, 37, ilat, ilon) = wind(ilat, ilon, 1, 6);
+              input_view(0, 38, ilat, ilon) = wind(ilat, ilon, 1, 7);
+            });
+      });
+}
+
+void ACE::run_impl(const double dt) {
+  // The main ACE run loop takes place here
+  std::cout << "ACE::run_impl: sc2ace remapping" << std::endl;
+  m_sc2ace_remapper->remap_fwd();
+  std::cout << "ACE::run_impl: ACE preprocessing" << std::endl;
+  preprocess();
+
+  // Detect model device and ensure tensor is on the same device
+  torch::Device model_device = torch::kCPU;
+  for (const auto& param : m_torch_data->module.parameters()) {
+    model_device = param.device();
+    break;  // We only need to check the first parameter
+  }
+  
+  std::cout << "ACE::run_impl: Model is on " 
+            << (model_device.is_cuda() ? "CUDA" : "CPU") 
+            << " device" << std::endl;
+
+  auto input_view = m_input_field.get_view<Real ****>();
+  
+  // we probably don't want to support dtype flexibility ... 
+  // let's force everyone to use kfloat32 and be done with it
+  // we should maybe even use lower precision...
+  // anyway, this will likely fail below (in the torch call, due to conflict with saved)
+  // auto torch_dtype = std::is_same<Real, double>::value ? torch::kFloat64 : torch::kFloat32;
+  // So, let's just force-use f32 for torch-specific ops for now
+  auto torch_dtype = torch::kFloat32;
+
+  // the tensor uses the same pointer as the big field view
+  auto input_tensor = torch::from_blob(
+    (void *)input_view.data(), {m_batch, m_in_ch, m_height, m_width},
+    torch::TensorOptions()
+        .dtype(torch_dtype)
+        .device(model_device)
+        .memory_format(torch::MemoryFormat::Contiguous));
+  
+  std::cout << "ACE::run_impl: ACE inference" << std::endl;
+  std::vector<torch::jit::IValue> inputs{input_tensor};
+  torch::Tensor output_tensor;
+  try { 
+    // TODO: this call creates and returns a tensor
+    // TODO: investigate how to make it recycle a view from above   
+    output_tensor = m_torch_data->module.forward(inputs).toTensor();
+  } catch(const c10::Error &e) {
+    std::cerr << "ACE::run_impl: Error during inference: " << e.what() << std::endl;
+    return;
+  }
+
+  // TODO: is there a way to avoid the forthcoming deep_copy?
```

Comment:
Presumably you could allocate the memory first and point pytorch tensor and Kokkos view to that, but you will probably need to be careful for different devices.